internal/pyxis/pyxis.go

```diff
@@ -408,12 +408,21 @@ func (p *pyxisClient) updateProject(ctx context.Context, certProject *CertProjec
 	return &newCertProject, nil
 }
 
-func (p *pyxisClient) createTestResults(ctx context.Context, testResults *TestResults) (*TestResults, error) {
+func (p *pyxisClient) createOrUpdateTestResults(ctx context.Context, testResults *TestResults, httpMethod string) (*TestResults, error) {
 	b, err := json.Marshal(testResults)
 	if err != nil {
 		return nil, fmt.Errorf("could not marshal test results: %w", err)
 	}
-	req, err := p.newRequestWithAPIToken(ctx, http.MethodPost, p.getPyxisURL(fmt.Sprintf("projects/certification/id/%s/test-results", p.ProjectID)), bytes.NewReader(b))
+
+	var pyxisURL string
+	switch httpMethod {
+	case http.MethodPatch:
+		pyxisURL = fmt.Sprintf("projects/certification/test-results/id/%s", testResults.ID)
+	default:
+		pyxisURL = fmt.Sprintf("projects/certification/id/%s/test-results", p.ProjectID)
+	}
```

Comment:
NIT: The function signature indicates it accepts _any_ HTTP method but then only actually executes MethodPatch or MethodPost. If a user passes in http.MethodGet, wouldn't this behave unexpectedly (falling back to http.MethodPost)?

Considered a nit because this is an internal function.

Reply:
It's a good call out, I was trying to avoid 2 separate functions, that are almost identical. Though based on your other comment about a net-new instance, maybe it should be 2 functions, and the `updateTestResults` struct is encapsulated in that method, like we have in other places. Though 2 functions for this will have alot of duplicated code, I'm fine either way though.

Reply:
The function returns an error, so we could potentially just case out the methods we want and return an error if the method is invalid.